### PR TITLE
Add GPU-accelerated testing

### PR DESCRIPTION
**Pull request by ianstenbit**
This PR is ~not yet ready for review~ now ready for review.

### CODE REVIEW

**Comment by ianstenbit** [comment]
GPU test workflow is now passing! @LukeWood

File: cloudbuild/cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  - -c
+  - |
+    set -u
+    set -e
+    set -x
+    gcloud container clusters get-credentials $_CLUSTER_NAME --zone $_CLUSTER_ZONE --project keras-team-test
+    job_name=$(kubectl create -f output.yaml -o name)
+    sleep 5
+    pod_name=$(kubectl wait --for condition=ready --timeout=10m pod -l job-name=${job_name#job.batch/} -o name)
+    kubectl logs -f $pod_name --container=train
```
Comment (sayakpaul) [inline cloudbuild/cloudbuild.yaml:71]: Might be cool use Kustomize for this purpose that lets you include a bunch manifests and just apply `kustomize build . -f -`.

**Comment by sayakpaul** [comment]
This is an amazing piece of work. I am referencing it for my own projects!

File: keras_cv/layers/preprocessing/grayscale.py [hunk 2]
```diff
@@ -50,6 +50,7 @@ class Grayscale(BaseImageAugmentationLayer):
     def __init__(self, output_channels=1, **kwargs):
         super().__init__(**kwargs)
         self.output_channels = output_channels
+        self.auto_vectorize = False
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/grayscale.py:53]: so grayscale doesn't work on GPU with auto_vectorize=True?

I think we should document this in a comment

**Review by LukeWood — CHANGES_REQUESTED** [review]
LGTM, but one comment.

File: keras_cv/layers/preprocessing/grayscale.py (hunk 2 above)
Comment (ianstenbit) [reply keras_cv/layers/preprocessing/grayscale.py:54]: Done.

**Review by LukeWood — APPROVED** [review]

File: cloudbuild/cloudbuild.yaml (hunk 1 above)
Comment (bhack) [reply cloudbuild/cloudbuild.yaml:71]: Yes I think that we could open a new ticket

**Pull request merged**